### PR TITLE
Add retry for Change Analysis and Geomaster

### PR DESCRIPTION
**Pull request by XiaoxuHU**
This PR is a following up one with [PR 645](https://github.com/Azure/Azure-AppServices-Diagnostics/pull/645), due to I unfolked repo and not able to push the last commit. 

This PR is for [WI 1657](https://dev.azure.com/app-service-diagnostics-portal/app-service-diagnostics-portal/_workitems/edit/1657) to add retry for other data providers

The latest change is I tried to tweak **RetryAsync** method in **RetryHelper** to pass func with either one param or no param to solve multi-thread issue last time Shekhar pointed out

### CODE REVIEW

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs [hunk 1]
```diff
@@ -568,22 +570,44 @@ public Task<T> InvokeDaasExtension<T>(string subscriptionId, string resourceGrou
...
+            {
+                HttpMethod = method,
+                Path = path,
+                QueryString = queryString,
+                Content = content,
+                ApiVersion = apiVersion,
+                CancellationToken = cancellationToken
+            };
+            
+            var task = RetryHelper.RetryAsync<R>(PerformHttpRequest<R>,param, "GeoMasterDataProvider", RequestId, _configuration.MaxRetryCount, _configuration.RetryDelayInSeconds * 1000);
```
Comment (ShekharGupta1988) [inline src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs:596]: nitpick : spacing between arguments. Just format the whole document :)

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs [hunk 2]
```diff
@@ -712,5 +736,14 @@ public override async Task<HealthCheckResult> CheckHealthAsync(CancellationToken
 
             return result ?? await base.CheckHealthAsync(cancellationToken);
         }
+
+        class GeoMasterRetryParam {
```
Comment (ShekharGupta1988) [inline src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs:740]: Does these needs to public classes or can they simply be internal or private?

**Review by ShekharGupta1988 — APPROVED** [review]

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs (hunk 1 above)
Comment (XiaoxuHU) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs:596]: Yes, will do

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs (hunk 2 above)
Comment (XiaoxuHU) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs:740]: I think for nested class default will be private(https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/nested-types)

image

Anyway, I can add private modifier for these class

**Pull request merged**